feat(ci): add support for shell-completion

ekifel opened the pull request:
Issue #525 

Initial support

alikhajeh1 commented:
Thanks!
Would it be possible to use the `--shell` flag? That enables us to have other flags in case we need it and do nicer error handling like https://github.com/cli/cli/blob/trunk/pkg/cmd/completion/completion.go#L56-L61 

Also, can we add https://github.com/spf13/cobra/blob/master/shell_completions.md#descriptions-for-completions and https://github.com/spf13/cobra/blob/master/shell_completions.md#mark-flags-as-required

ekifel commented:
@alikhajeh1, updated! Command's flags (with descriptions) shows by `ValidArgs: []string{"--", "-"}`, in one time flags worked without that (and don't understand why it worked), but I changed something and couldn't found how to bring it back, so used this `ValidArgs` construction. But if you think that it's weird and not obvious, I can set in `ValidArgs` every flag for every command.

And LMK please if you want me to specify something with flags

alikhajeh1 commented on cmd/infracost/completion.go:
```diff
@@ -0,0 +1,90 @@
+	
+	Bash:
+	
+		$ source <(infracost completion bash)
```
```suggestion
		$ source <(infracost completion --shell bash)
```

alikhajeh1 reviewed (commented):
Thanks @ekifel - I played with it on zsh and it's awesome! On bash, it doesn't show the descriptions, but it looks like that's how `gh` works too. Did you see the same issue?
Did it work ok on Powershell?

@aliscott would be great if you play with this too. Were the help instructions clear for you? Should we merge these (that were auto-generated) with the ones that `gh completion --help` includes?

aliscott commented on cmd/infracost/completion.go:
```diff
@@ -0,0 +1,90 @@
+	
+	fish:
+	
+		$ infracost completion fish | source
```
```suggestion
		$ infracost completion --shell fish | source
```

aliscott commented on cmd/infracost/output.go:
```diff
@@ -32,15 +32,11 @@ func outputCmd(cfg *config.Config) *cobra.Command {
 
-      infracost output --format json --path out*.json`,
+			infracost output --format json --path out*.json`,
+		ValidArgs: []string{"--", "-"},
```
What does this do?

ekifel replied:
Because without that autocompletion with flags doesn't work. Maybe exist another the way to do it, but I didn't find it

aliscott commented:
Works well for me. Should we make `--path` a required flag for `diff` and `breakdown` so it works with the autocompletion? If so, is there a way to show the current error message if it's missing since it has more context than the default "required flag(s) not set" message.

ekifel commented:
@aliscottб yes, I did so at first, but then I realized that if we make `--path` as a required flag, it will make impossible to use `--config-path`
